api/src/domain/tax-filings/taxFilingsInterfaceFactory.ts

```diff
@@ -34,15 +35,22 @@ export const taxFilingsInterfaceFactory = (apiTaxFilingClient: TaxFilingClient):
     const now = new Date(Date.now()).toISOString();
     let profileDataToReturn = request.userData.profileData;
 
-    if (naicsCode && taxCity) {
+    if (naicsCode) {
       profileDataToReturn = {
         ...request.userData.profileData,
         naicsCode: naicsCode,
+      };
+    }
+
+    if (taxCity) {
+      const record = await fetchMunicipalityByName(taxCity);
+      profileDataToReturn = {
+        ...profileDataToReturn,
         municipality: {
-          name: taxCity,
-          county: "",
-          id: "",
-          displayName: "",
+          name: record?.townName || profileDataToReturn.municipality?.name || "",
```

Comment:
[pebble] I'm thinking this logic isn't quite correct. If `record` fails to return and is undefined, and the user currently has an undefined municipality, then it will cascade down to saving it as `""`. It seems antithetical to our patterns to have a municipality that with all-empty fields. I think that if record is undefined then it should just stay set to what the municipality currently is - which means that if the municipality is currently undefined, it should stay undefined instead of becoming defined but with empty strings.

I think there are a lot of cases in our app where we check if a user has an industry by checking if municipality is undefined. If we set it defined, then those logics would break